### ISSUE TITLE
Auction Settlement

### ISSUE THREAD

**Issue by bh2smith** [gnosis/dex-contracts#55]
Contract should allow for State Transition as a result of auction results.

This would involve:
1. Input parameters (`oldStateRoot`, `newStateRoot`, `prices`, `fulfilmentVector`)
2. Check similar things to deposit and withdraw processing. For example 
       
- that current state root agrees 
- auction batch is an inactive slot
3. Emit event(s) with price & volume (i.e. settlement) information

**Comment by bh2smith** [comment]
```js
function applyAuction(
        uint slot,
        bytes32 _currStateRoot,
        bytes32 _newStateRoot,
        bytes32 _orderHash,
        bytes prices,
        bytes volume
) 
        public onlyOwner() 
{
       // Ensure current state root agrees
       // Ensure orderHashes match
       // Ensure previous auction slot has been applied
       // Ensure orderSlot is inactive

       // Do something with prices & volume
       stateRoots.push(_newStateRoot);        
       auctions[slot].appliedAccountStateIndex = stateIndex();
       emit StateTransition(TransitionType.Auction, stateIndex(), _newStateRoot, slot);
}
```

**Comment by bh2smith** [comment]
2.5 days spent

**Comment by fleupold** [comment]
As discussed on slack, we should combine prices and volume into a single vector. @bh2smith would you be able to support this?

**Comment by bh2smith** [comment]
Yes indeed, new proposed function input will be as follows, 

```js
event AuctionSettlement(
    uint auctionId,
    uint stateIndex,
    bytes32 stateHash,
    bytes solution  // representing [ prices | amounts ]
);
function applyAuction(
    uint slot,
    bytes32 _currStateRoot,
    bytes32 _newStateRoot,
    bytes32 _orderHash,
    bytes solution // representing [ prices | amounts ]
) 
{

    // Ensure current state root agrees
    // Ensure orderHashes match
    // Ensure previous auction slot has been applied
    // Ensure orderSlot is inactive

    // Do something with prices & volume
    stateRoots.push(_newStateRoot);        
    auctions[slot].appliedAccountStateIndex = stateIndex();
    
    emit AuctionSettlement(slot, stateIndex(), _newStateRoot, solution);
}
```

**Comment by bh2smith** [comment]
Closed by #65
